### ISSUE TITLE
Increase test coverage

### ISSUE THREAD

**Issue by Pennycook** [intel/code-base-investigator#36]
Our test coverage is currently only around 89%, which may be hiding some bugs.

A simple way to compute coverage statistics:
```sh
python -m coverage run -m unittest
python -m coverage report
```

**Comment by laserkelvin** [comment]
Should probably consider adding a coverage badge as well?

**Comment by Pennycook** [comment]
> Should probably consider adding a coverage badge as well?

Yes, good idea.  It would be good to have this as part of the CI.

**Comment by Pennycook** [comment]
As part of this issue, we should also refactor `bin/codebasin` so that the functionality can be tested by `unittest`.
